crates/recursor/src/recursor_dns_handle.rs

```diff
@@ -49,14 +50,15 @@ impl RecursorDnsHandle {
         recursion_limit: u8,
         security_aware: bool,
         do_not_query: Vec<IpNet>,
+        avoid_local_udp_ports: Arc<HashSet<u16>>,
     ) -> Result<Self, ResolveError> {
         // configure the hickory-resolver
         let roots: NameServerConfigGroup = roots.into();
 
         assert!(!roots.is_empty(), "roots must not be empty");
 
         debug!("Using cache sizes {}/{}", ns_cache_size, record_cache_size);
-        let opts = recursor_opts();
+        let opts = recursor_opts(Arc::clone(&avoid_local_udp_ports));
```

Comment:
Nit: I don't think we're very consistent about this but personally I prefer just `clone()`ing over the `Arc::clone()` thing.